### PR TITLE
Kuznetsov

### PR DESCRIPTION
**Pull request by Muao**
Fixed bugs after code review

### CODE REVIEW

File: kuznetsov/quizzes/src/main/java/com/kuznetsov/servlets/LoginServlet.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            services.getSavedCredentials().put(login, pwd);
+
+        } else {
+
+            if (services.getSavedCredentials().get(login).equals(pwd)) {
+
+                //FIXME insert quiz call
+
+            } else {
+                req.getSession().setAttribute("login", "wrong" );
```
Comment (dinamio) [inline kuznetsov/quizzes/src/main/java/com/kuznetsov/servlets/LoginServlet.java]: What if we registered user with "wrong" username?

File: kuznetsov/quizzes/quizzes.iml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module org.jetbrains.idea.maven.project.MavenProjectsManager.isMavenModule="true" type="JAVA_MODULE" version="4">
+  <component name="FacetManager">
```
Comment (dinamio) [inline kuznetsov/quizzes/quizzes.iml]: This file doesn't need to be in pr

File: kuznetsov/quizzes/src/main/java/com/kuznetsov/services/QuizServices.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    private String pwd;
+    private String login;
+    private Map<String, String> savedCredentials = new HashMap<>();
+
+    public Map<String, String> getSavedCredentials() {
+        return savedCredentials;
+    }
+
+    public void setSavedCredentials(String login, String pwd) {
```
Comment (dinamio) [inline kuznetsov/quizzes/src/main/java/com/kuznetsov/services/QuizServices.java]: I think this login logic and credentials logic need to be in separate service

**Comment by Muao** [comment]
Home work #4 is implemented

**Review by dinamio — CHANGES_REQUESTED** [review]
Please, look at comments

File: kuznetsov/quizzes/quizzes.iml (hunk 2 above)
Comment (Muao) [reply kuznetsov/quizzes/quizzes.iml]: Without this file, you can not run the project without first configuring it. I had a comment on this last time.

File: kuznetsov/quizzes/src/main/java/com/kuznetsov/services/QuizServices.java (hunk 3 above)
Comment (Muao) [reply kuznetsov/quizzes/src/main/java/com/kuznetsov/services/QuizServices.java]: Ok, will do.

File: kuznetsov/quizzes/src/main/java/com/kuznetsov/servlets/LoginServlet.java (hunk 1 above)
Comment (Muao) [reply kuznetsov/quizzes/src/main/java/com/kuznetsov/servlets/LoginServlet.java]: It's very old wrong realization. I have no no thoughts, how it comes to pull request. Last realization of LoginServlet.java here: https://github.com/dinamio/hillelee/blob/kuznetsov/kuznetsov/quizzes/src/main/java/com/kuznetsov/servlets/LoginServlet.java

**Comment by Muao** [comment]
Made changes recommended in the comments

File: kuznetsov/quizzes/src/main/java/com/kuznetsov/enteties/Credentials.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
+package enteties;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Credentials {
```
Comment (dinamio) [inline kuznetsov/quizzes/src/main/java/com/kuznetsov/enteties/Credentials.java:6]: Better to separate savedCredentials logic into separate service, to not mix logic with entity

**Review by dinamio — APPROVED** [review]
Look good!

File: kuznetsov/quizzes/src/main/java/com/kuznetsov/enteties/Credentials.java (hunk 4 above)
Comment (Muao) [reply kuznetsov/quizzes/src/main/java/com/kuznetsov/enteties/Credentials.java:6]: Understood. Thank you.

**Pull request merged**